### PR TITLE
Add Preserve Data

### PR DESCRIPTION
**Pull request by NatiDeme**
# Changes in this PR:
- No linter errors
- Data saved on JSON files
- On startup, data is loaded from files.
-  Missing files are handled.

### CODE REVIEW

File: save_data.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    File.write('./persons.json', JSON.generate(people_to_save))
+  end
+
+  def save_rentals(rentals)
+    rentals_to_save = []
+    rentals.each do |rental|
+      rentals_to_save << { 'date' => rental.date,
+                           'person_name' => rental.person.name,
+                           'book_title' => rental.book.title, 'person_id' => rental.person.id }
+    end
```
Comment (dasileker) [inline save_data.rb]: - [x] The  'date' in your save_rentals is not defined it returns an error while saving the rentals.

Capture d’écran 2022-08-31 103604

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @NatiDeme,** 

### **Good job so far!**
There are some issues that you still need to work on to go to the next project but you are almost there!

## **Highlights**

- Linters are green ✅.
- Commit are consistent✅.
- PR is very good ✅.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: create_item.rb [hunk 2]
```diff
@@ -78,7 +84,6 @@ def create_book
     author = gets.chomp
 
     new_book(title, author)
-
     puts('Book created successfully')
     run
   end
```
Comment (Bondok6) [inline create_item.rb:89]: - [OPTIONAL] Great job implementing the changes by the previous reviewer 👌 but I noticed that the new_book doesn't store in the books.json file, I think you forgot to use `save_book(@book)`, so kindly fix that. I'll mark it as optional but I strongly recommend fixing this issue 💯👌
```
  def new_book(title, author)
    new_book = Book.new(title, author)
    @book << new_book
    save_book(@book)
  end
```
Your Code:  
```
  def new_book(title, author)
    new_book = Book.new(title, author)
    @book << new_book
  end
```

**Review by Bondok6 — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

## Hi @NatiDeme  🙋‍♂️, 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

TO HIGHLIGHT:

- Your project is professional ✅
- Your repo is professional ✅
- Well structured files ✅
- No Linter errors ✅


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 




## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**